skore/src/skore/item/cross_validate_item.py

```diff
@@ -0,0 +1,232 @@
+"""CrossValidationItem class.
+
+This class represents the output of a cross-validation workflow.
+"""
+
+from __future__ import annotations
+
+import contextlib
+import hashlib
+from typing import TYPE_CHECKING, Any
+
+import altair
+import numpy
+
+from skore.item.item import Item
+from skore.item.media_item import MediaItem
+
+if TYPE_CHECKING:
+    import numpy
+    import sklearn.base
+
+
+def plot_cross_validation(cv_results: dict) -> altair.Chart:
+    """Plot the result of a cross-validation run.
+
+    Parameters
+    ----------
+    cv_results : dict
+        The output of scikit-learn's cross_validate function.
+
+    Returns
+    -------
+    altair.Chart
+        A plot of the cross-validation results
+    """
+    import altair
+    import pandas
+
+    _cv_results = cv_results.copy()
+
+    with contextlib.suppress(KeyError):
+        del _cv_results["indices"]
+        del _cv_results["estimator"]
+
+    df = (
+        pandas.DataFrame(_cv_results)
+        .reset_index(names="split")
+        .melt(id_vars="split", var_name="metric", value_name="score")
+    )
+
+    input_dropdown = altair.binding_select(
+        options=df["metric"].unique().tolist(), name="Metric: "
+    )
+    selection = altair.selection_point(
+        fields=["metric"], bind=input_dropdown, value="test_score"
+    )
+
+    return (
+        altair.Chart(df, title="Cross-validation scores per split")
+        .mark_bar()
+        .encode(
+            altair.X("split:N").axis(
+                title="Split number",
+                labelAngle=0,
+            ),
+            altair.Y("score:Q").axis(
+                title="Score",
+                titleAngle=0,
+                titleAlign="left",
+                titleX=0,
+                titleY=-5,
+                labelLimit=300,
+            ),
+            tooltip=["metric:N", "split:N", "score:Q"],
+        )
+        .interactive()
+        .add_params(selection)
+        .transform_filter(selection)
+        .properties(
+            width=500,
+            height=200,
+            padding=15,
+            autosize=altair.AutoSizeParams(type="pad", contains="padding"),
+        )
+    )
+
+
+def _hash_numpy(array: numpy.ndarray) -> str:
+    """Compute a hash string from a numpy array.
+
+    Parameters
+    ----------
+    array : numpy array
+        The numpy array whose hash will be computed.
+
+    Returns
+    -------
+    hash : str
+        A hash corresponding to the input array.
+    """
+    return hashlib.sha256(array.tobytes()).hexdigest()
```

Comment:
Can [`struct.pack`](https://docs.python.org/3/library/struct.html#struct.pack) be used here to make this hash function generic ?

Reply:
struct.pack seems to only accept ints. In the end, it should be enough to simply convert X and y to numpy arrays.